src/context/context.ts

```diff
@@ -67,9 +67,12 @@ export class Context {
   }
```

Comment:
Thess lines are possibly broken:

https://github.com/harttle/liquidjs/pull/285/files#diff-05ca53b312ab10fd3af29de11c6b119e8a3e13f8fe2ba6c9353977498da1ed41R43-R44

`.reduce` is executed syncly, and the following case will fail:

```javascript
const val = await liquid.parseAndRender('{{ foo.bar }}', { foo: Promise.resolve({bar: 'BAR'}) })
expect(var).to.equal('BAR')
```

I suggest replace the `.reduce()` with `for...of` and `yield readProperty()`.

Reply:
Does it mean that getFromScope has to be generator function? i think this can be a breaking change.\

I tried the above test and it passed.

Reply:
It's OK to leave it alone. I'll handle this latter.